cmd/routed-eni-cni-plugin/driver/driver.go

```diff
@@ -162,35 +170,58 @@ func (createVethContext *createVethPairContext) run(hostNS ns.NetNS) error {
 	var maskLen int
 	var addr *netlink.Addr
 	var defNet *net.IPNet
+	var family int
 
-	if createVethContext.v4Addr != nil {
-		gw = net.IPv4(169, 254, 1, 1)
+	if createVethContext.ipAddr.IP.To4() != nil {
+		gw = net.IPv4(169, 254, 1, byte(createVethContext.index)+1)
```

Comment:
let's have a helper function for this as well.

bonus point, group those helpful function that define our network setup for given card/index in a central place.

Reply:
Updated this